### DISCUSSION TITLE
Controller and other weirdness on steamdeck

### DISCUSSION THREAD

**Discussion by TheFeri** [Heroic-Games-Launcher/HeroicGamesLauncher#3568] [Q&A]
I have a PS4 controller connected via bluetooth. When I want to play games using heroic in desktop mode the game just doesn't recognizes the input but knows its there, and what I mean by that is that no matter what I press nothing happens but if I disconnect the controller the game acknowledges it's gone and wants it back.
I saw a similar post here saying closing steam and then launching heroic/the game should work, it did not. Then I tried adding as a non-steam game(using the feature in heroic) exactly the same issues. Then I wanted to try gaming mode, it defintely should work there... the game doesn't even launch in gaming mode...

So at this point... what should I do? I Also tried big picture mode but that defaults to the decks screen making it pointelss(because obviously I'm docked with a monitor that's why I'm using desktop mode)

**Comment by evil77hoelle** [comment]
Yesterday i was looking for a way to use my PS4 controller wit HGL on fedora 43.
The easiest fix i found (and i do not understand why this fix is not more prominently advertised);

To enable PS4 controller in game:

(for one game)Settings - ADVANCED - Environment Variables -
#or#
(globally) Settings - Game Defaults - ADVANCED - Environment Variables -

Variable Name =  PROTON_PREFER_SDL
Value = 1

Maybe this helps on the steamdeck to, if so plz comment.
